### PR TITLE
Metrics for shared channels

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR adds additional metrics for Shared Channels.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56777

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/enterprise/metrics/metrics.go [hunk 1]
```diff
@@ -927,6 +936,90 @@ func New(ps *platform.PlatformService, driver, dataSource string) *MetricsInterf
...
+		[]string{"remote_id"},
+	)
+	m.Registry.MustRegister(m.SharedChannelsSyncCount)
+
+	m.SharedChannelsTaskInQueueHistogram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace:   MetricsNamespace,
+			Subsystem:   MetricsSubsystemSharedChannels,
+			Name:        "task_in_queue_duration",
+			Help:        "Duration tasks spend in queue (millis)",
```
Comment (lieut-data) [inline server/enterprise/metrics/metrics.go:958]: Can we emit these in (fractional) seconds? As per https://prometheus.io/docs/practices/naming/
> ...should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers). See below for a list of base units.

**Review by lieut-data — CHANGES_REQUESTED** [review]
This looks great, @wiggin77! Just one ask to switch to emitting as seconds to Prometheus below.

File: server/enterprise/metrics/metrics.go (hunk 1 above)
Comment (wiggin77) [reply server/enterprise/metrics/metrics.go]: Done

**Review by lieut-data — APPROVED** [review]
Thanks, @wiggin77!

**Review by sbishel — APPROVED** [review]

**Pull request merged**